src/base/bittorrent/sessionimpl.cpp

```diff
@@ -683,16 +683,25 @@ SessionImpl::SessionImpl(QObject *parent)
     // start embedded tracker
     enableTracker(isTrackerEnabled());
 
-    prepareStartup();
-
     m_updateTrackersFromURLTimer = new QTimer(this);
     m_updateTrackersFromURLTimer->setInterval(24h);
-    connect(m_updateTrackersFromURLTimer, &QTimer::timeout, this, &SessionImpl::updateTrackersFromURL);
-    if (isAddTrackersFromURLEnabled())
+    connect(m_updateTrackersFromURLTimer, &QTimer::timeout, this, [this]()
     {
         updateTrackersFromURL();
+    });
+
+    if (isAddTrackersFromURLEnabled())
+    {
+        updateTrackersFromURL([this]()
+        {
+            prepareStartup();
+        });
```

Comment:
I wouldn't call it a good idea to perform tasks sequentially that are ideal candidates for parallel execution. It seems that the issue requires a slightly more advanced solution. (Of course, if it's worth the effort. #23508 doesn't seem to have a large enough coverage so we could consider it as a known limitation rather than a bug.)